### DISCUSSION TITLE
Compile styles in a variable in the browser

### DISCUSSION THREAD

**Discussion by Garito** [tailwindlabs/tailwindcss#10761] [Help]
Hi!
It is possible to compile the contents of a variable?
So giving this:
```js
const css = "
  @tailwind base;
  @tailwind components;
  @tailwind utilities;

  .class {
    @apply w-full;
  }
"
```
I would like to get the compiled css and store it in another variable
Is this possible in the browser?
How?
Thanks

**Comment by wongjn** [comment]
You could try compiling the CSS via [the CDN](https://tailwindcss.com/docs/installation/play-cdn):
```html
<script src="https://cdn.tailwindcss.com"></script>
<style type="text/tailwindcss">
  .class {
    @apply w-full;
  }
</style>
```
Then you can use JavaScript to find the `<style>` element that the CDN generated.

**Reply by Garito** [reply]
No other option?
We are building a SaaS and this solution seems more a hack that a proper solution
But thanks for your time

**Reply by wongjn** [reply]
You could send your CSS to an endpoint or a Web Container that runs Tailwind via the PostCSS API.

**Reply by Garito** [reply]
More hacks?
Are you a tailwind developer?
Would like a proper solution since we plan to use this feature full scale...
I appreciate you effort even when I hesitate to build your suggestions. I hope you don't take it in a bad way and you can empathize with my position here

**Reply by wongjn** [reply]
Could I ask what you would consider a "proper solution" please, or how you consider the previous suggestions as "hacks"? Perhaps if we had some concrete criteria, we could look for a solution that you would consider "proper"!

**Reply by Garito** [reply]
If you are (a tailwind developer) I would appreciate you tell me there is no canonical way to do what we need and that we have to go for the hacks

**Reply by Garito** [reply]
A proper solution will be a pure javascript way to call tailwind cli or postcss cli programatically

**Reply by wongjn** [reply]
The CLIs call the programmatic APIs anyway, so you could skip that step and run Tailwind via the PostCSS programmatic API.

**Reply by Garito** [reply]
Can you show how?
I've tried but a fs error raises:
```
Uncaught ReferenceError: fs$4 is not defined
```
This is what I've tried:
``` js
const compiledCss = await postcss([
      tailwindcss({ content: [], theme: { extend: {} }, plugins: [] }),
      autoprefixer
    ]).process(css);
```

**Reply by wongjn** [reply]
Yup, that's pretty much it:

```js
const postcss = require('postcss');
const tailwindcss = require('tailwindcss');

const tailwindConfig = {
  content: [{ raw: getMarkup(), extension: 'html' }],
  theme: {
    // …
  },
};
const resultCSS = await postcss([tailwindcss(tailwindConfig)]).process(sourceCSS);
```

**Reply by Garito** [reply]
Is this a browser compatible solution?
Isn't ```require``` a node way to import libs?

**Reply by wongjn** [reply]
> Is this a browser compatible solution?

Probably not as-is.

> Isn't `require` a node way to import libs?

It is, yes.

**Reply by Garito** [reply]
That's what I thought
I would write a request feature then
Thanks for your help
